sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -376,6 +381,21 @@ def __init__(
     def __len__(self):
         return self.size
 
+    def _get_encryption_data_request(self):
+        # Save current request cls
+        download_cls = self._request_options.pop('cls', None)
+        # Adjust cls for get_properties
+        self._request_options['cls'] = deserialize_blob_properties
```

Comment:
Modifying `self` makes this object not threadsafe - is there any possibility this method could be run concurrently?